op.c

```diff
@@ -15268,23 +15284,34 @@ Perl_rcpv_new(pTHX_ const char *pv, STRLEN len, U32 flags) {
 
     PERL_UNUSED_CONTEXT;
 
+    /* Musn't use both at the same time */
+    assert((flags & (RCPVf_NO_COPY|RCPVf_USE_STRLEN))!=
+                    (RCPVf_NO_COPY|RCPVf_USE_STRLEN));
+
     if (!pv && (flags & RCPVf_NO_COPY) == 0)
         return NULL;
 
     if (flags & RCPVf_USE_STRLEN)
         len = strlen(pv);
 
-    rcpv = (RCPV *)PerlMemShared_malloc(sizeof(struct rcpv) + len + 1);
+#if DEBUGGING
+    assert(len || (flags & RCPVf_ALLOW_EMPTY));
+#endif
```

Comment:
The assert() will only typically apply when DEBUGGING is defined, so the #ifdef is unnecessary.

Reply:
Yeah, that was a stupid. Originally I had a warn in there, I just forgot to remove the #if when I changed it to an assert.

Fixed.